### PR TITLE
[buteo-sync-plugin-caldav] Use lcCalDav* logging categories instead of Buteo LOG_* macros. JB#53712

### PR DESCRIPTION
**Pull request by blammit**

### CODE REVIEW

File: src/authhandler.cpp [hunk 1]
```diff
@@ -66,23 +66,23 @@ AuthHandler::AuthHandler(Accounts::Manager *manager, const quint32 accountId, co
 
 bool AuthHandler::init()
 {
-    FUNCTION_CALL_TRACE;
+    FUNCTION_CALL_TRACE(lcCalDav);
```
Comment (dcaliste) [inline src/authhandler.cpp:69]: In the corresponding Buteo PR, the `FUNCTION_CALL_TRACE` macro is used with the trace variant of the logging category. Why not in CalDAV also ?

File: src/caldavclient.cpp [hunk 2]
```diff
@@ -36,12 +36,13 @@
 #include <QNetworkReply>
 #include <QDateTime>
 #include <QtGlobal>
+#include <QStandardPaths>
```
Comment (dcaliste) [inline src/caldavclient.cpp:39]: Is it a leftover of something ? I don't see where it is need in the changes.

**Review by dcaliste — CHANGES_REQUESTED** [review]
Very nice transition to Qt logging categories. Thank you.

I put a long time ago a line like `qputenv("MSYNCD_LOGGING_LEVEL", "8");` in tests/notebooksyncagent/tst_notebooksyncagent.cpp to remember about how to obtain full debug in tests. Maybe it should be updated also in this PR.

File: src/authhandler.cpp (hunk 1 above)
Comment (blammit) [reply src/authhandler.cpp]: Yes this should also have a trace variant, thanks for picking that up. Fixed now.

File: src/caldavclient.cpp (hunk 2 above)
Comment (blammit) [reply src/caldavclient.cpp:39]: This file was depending on an indirect include of QStandardPaths via LogMacros.h, which has now been removed.

**Comment by blammit** [comment]
> I put a long time ago a line like `qputenv("MSYNCD_LOGGING_LEVEL", "8");` in tests/notebooksyncagent/tst_notebooksyncagent.cpp to remember about how to obtain full debug in tests. Maybe it should be updated also in this PR.

Thanks for pointing that out, I've remove that now.

**Review by chriadam — APPROVED** [review]

**Pull request merged**